dsmr_parser/telegram_specifications.py

```diff
@@ -181,3 +181,19 @@
         obis.INSTANTANEOUS_CURRENT_L3: CosemParser(ValueParser(Decimal)),
     }
 }
+
+
+Q3D = {
```

Comment:
Can you provide the specification for the protocol profile this meter is using? (A URL would be fine.)
(Maybe the name prefixes LUXEMBOURG_ is awkward, and we find out this is a more generally used item and can fix the names later on.)

Reply:
Here is some documentation I found:
[EasyMeter Q3D Manual (german) ](https://www.easymeter.com/downloads/products/zaehler/Q3D/Easymeter_Q3D_DE_2016-06-15.pdf)
- type code on page 8
- D0-Specs on page 20

[Implementation notes for a Q3D reader](http://www.lab4inf.fh-muenster.de/lab4inf/docs/Embedded-Software/SmartMeter_UART.pdf)
- telegram analysis starting slide 15

[OBIS code table](https://www.promotic.eu/en/pmdoc/Subsystems/Comm/PmDrivers/IEC62056_OBIS.htm)

The latter just calls "1.8.0" / "2.8.0" "positive" / "negative" active energy total. 

So for my taste these 
- ELECTRICITY_IMPORTED_TOTAL = r'\d-\d:1\.8\.0.+?\r\n'
- LUXEMBOURG_ELECTRICITY_USED_TARIFF_GLOBAL = r'\d-\d:1\.8\.0.+?\r\n'  # Total imported energy register (P+)
- LUXEMBOURG_ELECTRICITY_DELIVERED_TARIFF_GLOBAL = r'\d-\d:2\.8\.0.+?\r\n'  # Total exported energy register (P-)
- SWEDEN_ELECTRICITY_USED_TARIFF_GLOBAL = r'\d-\d:1\.8\.0.+?\r\n'  # Total imported energy register (P+)
- SWEDEN_ELECTRICITY_DELIVERED_TARIFF_GLOBAL = r'\d-\d:2\.8\.0.+?\r\n'  # Total exported energy register (P-)
could all be simplified to 
- ELECTRICITY_IMPORTED_TOTAL = r'\d-\d:1\.8\.0.+?\r\n'
- ELECTRICITY_EXPORTED_TOTAL = r'\d-\d:2\.8\.0.+?\r\n'

But to not break things I would just add ELECTRICITY_EXPORTED_TOTAL and use them in the Q3D definition, instead of the LUXEMBOURG_ ones.

Reply:
This is the last thing we need to do, I think. The problem is that the parser can not distinguish between ELECTRICITY_IMPORTED_TOTAL, LUXEMBOURG_ELECTRICITY_USED_TARIFF_GLOBAL, SWEDEN_ELECTRICITY_USED_TARIFF_GLOBAL, as they are exactly the same regarding the matching pattern used.

Same for LUXEMBOURG_ELECTRICITY_DELIVERED_TARIFF_GLOBAL, SWEDEN_ELECTRICITY_DELIVERED_TARIFF_GLOBAL, ELECTRICITY_EXPORTED_TOTAL.

To unbreak things, we have to implement your simplified proposal. You want to give it a go?

Reply:
I do get your point, that the same code has different labels per different specifications. When I saw this for the first time, I tought it was a way to be more specific per each country using the codes slightly differently. But I could be wrong.

Since we excatly give the parser just one spec to run with (e.g. 5L), there is no harm for the parser. He only looks at that 5L spec. As long as ELECTRICITY_IMPORTED_TOTAL, LUXEMBOURG_ELECTRICITY_USED_TARIFF_GLOBAL, SWEDEN_ELECTRICITY_USED_TARIFF_GLOBAL don't show up in one spec, this should be fine. I have also not seen any issues reported on this.

So much to the conceptual part. 

What remains is cosmetics to deduplicate the SWEDEN/LUXEMBOURG labels. I would still suggest that I do that in a separate pull request, to track issues on possible downstrem dependecies.

Reply:
If we are really bold, we may even go a bit further and see if we could remove the different specs entirely and unify them. This would remove the need to know which spec you need, remove the parameter/option and just deliver data based on a standard OBIS interpretation. 

This would only work if the country/vendor specific codes are optional. 

The parser could also "warn" if it discovers labels that are not part of the spec and not drop them, but forward them with "numeric" labels to be picked up downstream or enhancement of our spec here.

It might well be a breaking change, so I would suggest do that in a new branch. Also we would need to have folks at least from Sweden and Luxembourg to commit some time for testing, as I only have access to my two little boxes here.

Again, just to be be clear, these are just thoughts and goes well beyond this pull request.